src/main/java/org/javarosa/xpath/expr/XPathFormatDateForCalendarFunc.java

```diff
@@ -22,29 +23,41 @@ public XPathFormatDateForCalendarFunc(XPathExpression[] args) throws XPathSyntax
     }
 
     @Override
+    protected void validateArgCount() throws XPathSyntaxException {
+        if (args.length < 2 || args.length > 3) {
+            throw new XPathArityException(name, "2 or 3 arguments", args.length);
+        }
+    }
+
+    @Override
     public Object evalBody(DataInstance model, EvaluationContext evalContext, Object[] evaluatedArgs) {
-        return formatDateForCalendar(evaluatedArgs[0], evaluatedArgs[1]);
+        String formatString = null;
```

Comment:
I could see the value there. In theory if that was a bad 'default' format for some future calendar, that would prevent that calendar from being able to differentiate between a specific format request and the default without modifying this core method, though, so I think this is a slightly safer fallback.